lib/async.js

```diff
@@ -225,6 +225,7 @@ module.exports = function resolve(x, options, callback) {
         }
     }
     function loadNodeModules(x, start, cb) {
+        opts.request = x;
```

Comment:
Can we try to find a way to avoid mutating `opts` here?

Reply:
There are a few options:
- I can pass `x` as optional third parameter of `nodeModulesPaths`
- I can copy `opts` using `Object.assign` (Node 4+)
- I can keep mutating `opts`, but copy it at the start of `resolve` / `resolveSync`

Also note that `opts` is already mutated in the codebase: https://github.com/browserify/resolve/blob/master/lib/sync.js#L42

Do you have a preference?

Reply:
Done 👍